work_queue/src/work_queue.c

```diff
@@ -5533,6 +5567,10 @@ struct work_queue_task *work_queue_wait_internal(struct work_queue *q, int timeo
 {
 	int events = 0;
 
+	q->stats->prev_time_application = q->stats->time_application;
```

Comment:
Instead of having prev_* fields,  another field to struct work_queue,  stats_prev. Here simply do a memcpy.